### PR TITLE
Update MLflow artifact viewer to show correct recommendation when loa…

### PR DESCRIPTION
**Pull request by mingyu89**
Spark models can’t be evaluated using Spark UDFs, so we shouldn’t show the spark udf code snippet if the model has the pyfunc flavor with loader module config mlflow.spark.

Before the change, it's showing
Screen Shot 2022-09-30 at 12 58 06 AM

After the change(made the same change in db repo and tested)
Screen Shot 2022-10-03 at 12 59 39 PM

And the recommendation can actually do inference and return value
Screen Shot 2022-09-30 at 12 49 12 AM

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [x] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by jinzhang21 — COMMENTED** [review]
I thought running Spark model as UDF is supported? https://www.mlflow.org/docs/latest/python_api/mlflow.pyfunc.html#mlflow.pyfunc.spark_udf

**Comment by dbczumar** [comment]
> I thought running Spark model as UDF is supported? https://www.mlflow.org/docs/latest/python_api/mlflow.pyfunc.html#mlflow.pyfunc.spark_udf

Unfortunately, we can't embed a Spark model in a Spark UDF. This becomes Spark squared - i.e. a distributed model being passed to individual workers. It doesn't work, and there isn't any need for it (using PySpark for inference is sufficient).

File: mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactLoggedModelView.js [hunk 1]
```diff
@@ -195,6 +207,9 @@ class ShowArtifactLoggedModelView extends Component {
   }
 
   renderPyfuncCodeSnippet() {
+    if (this.state.loader_module === 'mlflow.spark') {
+      return this.renderMlflowSparkCodeSnippet();
```
Comment (dbczumar) [inline mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactLoggedModelView.js:211]: Can we also indicate that users can reload the model via `mlflow.pyfunc.load_model()` and perform single-node predictions via pyfunc predict (https://github.com/mlflow/mlflow/blob/97506b50fb32d0499a8239b69a10bc4812525f84/mlflow/spark.py#L896-L925)?

File: mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactLoggedModelView.js [hunk 2]
```diff
@@ -309,6 +324,64 @@ class ShowArtifactLoggedModelView extends Component {
...
+          <div css={styles.item}>
+            <Paragraph
+              dangerouslySetAntdProps={{
+                copyable: { text: this.mlflowSparkCodeText(modelPath) },
+              }}
+            >
+              <pre style={{ wordBreak: 'break-all', whiteSpace: 'pre-wrap', marginTop: 10 }}>
+                <div className='code'>
+                  <span className='code-keyword'>import</span> mlflow{`\n`}
+                  logged_model = <span className='code-string'>{`'${modelPath}'`}</span>
```
Comment (dbczumar) [inline mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactLoggedModelView.js:349]: Looks like we're duplicating a lot of code here. Can we factor it out?

File: mlflow/server/js/src/i18n/default/en.json [hunk 3]
```diff
@@ -839,6 +843,10 @@
     "defaultMessage": "Parameters",
     "description": "Table title text for parameters table in the model comparison page"
   },
+  "bUNg0K": {
+    "defaultMessage": "Perform inference via model.transform()",
+    "description": "Code comment which states on how we can perform inference"
+  },
```
Comment (dbczumar) [inline mlflow/server/js/src/i18n/default/en.json:849]: Nit:

```suggestion
  "bUNg0K": {
    "defaultMessage": "Perform inference via model.transform()",
    "description": "Code comment which states how we can perform SparkML inference"
  },
```

File: mlflow/server/js/src/i18n/default/en.json [hunk 4]
```diff
@@ -451,6 +451,10 @@
     "defaultMessage": "Show diff only",
     "description": "Toggle text that determines whether to show diff only in the model\n                      comparison page"
   },
+  "H8/czZ": {
+    "defaultMessage": "Load model",
+    "description": "Code comment which states how to load model"
```
Comment (dbczumar) [inline mlflow/server/js/src/i18n/default/en.json:456]:
```suggestion
    "description": "Code comment which states how to load a SparkML model"
```

File: mlflow/server/js/src/i18n/default/en.json (hunk 3 above)
Comment (mingyu89) [reply mlflow/server/js/src/i18n/default/en.json:849]: This file is auto-generated. I'll modify the source.

File: mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactLoggedModelView.js (hunk 2 above)
Comment (mingyu89) [reply mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactLoggedModelView.js:349]: The duplicate code is the three lines of code - make predictions, import, assign log model
Screen Shot 2022-10-03 at 11 16 44 AM

And it's duplicating only a small part of 1/2 of renderPyfuncCodeSnippet. I feel it's cleaner to keep them separate, how you feel?

Comment (mingyu89) [reply mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactLoggedModelView.js:349]: Oh I'll have to add back "pyfunc.load_model". Then that's a lot of duplicate code. I'll try to simplify.

**Comment by mingyu89** [comment]
@dbczumar Please take another look and let me know if there's anything I need to change.

File: mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactLoggedModelView.js (hunk 2 above)
Comment (mingyu89) [reply mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactLoggedModelView.js]: Only moved the pandas frame part to a method:(

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @mingyu89 !

**Pull request merged**